src/main/java/hexgui/gui/GuiBoard.java

```diff
@@ -0,0 +1,701 @@
+      }
+    }
+    repaint();
+  }
+
+  /** Clear swap marks */
+  public void clearSwapPlayed() {
+    for (int x = 0; x < m_field.length; x++) {
+      m_field[x].clearAttributes(GuiField.SWAP_PLAYED);
+    }
+    repaint();
+  }
+
+  /** Add swap mark to all pieces on the board (hopefully there is exactly one of them */
+  public void markSwapPlayed() {
+    for (int x = 0; x < m_field.length; x++) {
+      HexPoint p = m_field[x].getPoint();
+      if (p.is_cell() && m_field[x].getColor() != HexColor.EMPTY) {
+        m_field[x].setAttributes(GuiField.SWAP_PLAYED);
+      }
+    }
+    repaint();
+  }
+
+  /** Sets the given point's alpha color. */
+  public void setAlphaColor(HexPoint point, Color color) {
+    GuiField f = getField(point);
+    if (f != null) {
+      f.setAlphaColor(color);
+      repaint();
+    }
+  }
+
+  public void setAlphaColor(HexPoint point, Color color, float blend) {
+    GuiField f = getField(point);
+    if (f != null) {
+      f.setAlphaColor(color, blend);
+      repaint();
+    }
+  }
+
+  /** Returns the point's alpha color; null if it is 'swap-sides' or resign or similar. */
+  public Color getAlphaColor(HexPoint point) {
+    GuiField f = getField(point);
+    if (f != null) {
+      return f.getAlphaColor();
+    } else {
+      return null;
+    }
+  }
+
+  /** Sets the given point's text. */
+  public void setText(HexPoint point, String str) {
+    getField(point).setText(str);
+    repaint();
+  }
+
+  /** Sets whether this cell is selected. */
+  public void setSelected(HexPoint point, boolean selected) {
+    getField(point).setSelected(selected);
+    repaint();
+  }
+
+  /** Check if the board is full */
+  public boolean isBoardFull() {
+    for (int x = 0; x < m_field.length; x++) {
+      if (m_field[x].getColor() == HexColor.EMPTY) return false;
+    }
+    return true;
+  }
+
+  /** Count the number of pieces on the board */
+  public int numberOfPieces() {
+    int count = 0;
+    for (int x = 0; x < m_field.length; x++) {
+      HexPoint point = m_field[x].getPoint();
+      if (point == HexPoint.NORTH
+          || point == HexPoint.EAST
+          || point == HexPoint.SOUTH
+          || point == HexPoint.WEST) {
+        continue;
+      }
+      if (m_field[x].getColor() != HexColor.EMPTY) {
+        count++;
+      }
+    }
+    return count;
+  }
+
+  /** Change the pieces' colors without moving them. This is only used in Y. */
+  public void swapColors() {
+    for (int x = 0; x < m_field.length; x++) {
+      HexPoint point = m_field[x].getPoint();
+      if (point == HexPoint.NORTH
+          || point == HexPoint.EAST
+          || point == HexPoint.SOUTH
+          || point == HexPoint.WEST) {
+        continue;
+      }
+      HexColor color = m_field[x].getColor();
+      m_field[x].setColor(color.otherColor());
+    }
+  }
+
+  /** Change the pieces' colors and move them. This is only used in Hex. */
+  public void swapPieces() {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 11. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code blocks on lines 482-491, lines 492-502 into separate methods.